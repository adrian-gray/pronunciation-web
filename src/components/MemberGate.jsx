import React from 'react'

import {
  Typography,
  withStyles
} from '@material-ui/core'

const styles = (theme) => ({
  headspace: theme.headspace
})

const MemberGate = (props) => {
  let display

  if (props.userAuth) {
    display = props.content
  } else {
    display = (
      <Typography variant='subheading' className={props.classes.headspace} gutterBottom>
        {'Sorry, this activity is for members only.'}
      </Typography>
    )
  }

  return display
}

export default withStyles(styles)(MemberGate)
