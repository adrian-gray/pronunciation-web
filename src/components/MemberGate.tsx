import React from "react";

export default props => {
  let display;

  if (props.userAuth) {
    display = props.content;
  } else {
    display = (
      <div>
        <h4 className="head-space">
          {"Sorry, this interactive activity is for members only."}
        </h4>
        <div className="blocked">{props.content}</div>
      </div>
    );
  }

  return display;
};
