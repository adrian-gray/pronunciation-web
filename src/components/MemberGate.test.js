import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import MemberGate from "./MemberGate";

afterEach(cleanup);

test("MemberGate authorised", async () => {
  const props = {
    userAuth: false
  };

  const container = shallow(<MemberGate {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
