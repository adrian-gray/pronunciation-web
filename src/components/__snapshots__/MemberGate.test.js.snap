// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberGate authorised 1`] = `undefined`;
